canvas.py

```diff
@@ -0,0 +1,94 @@
+from random import random
+from kivy.app import App
+from kivy.uix.widget import Widget
+from kivy.uix.button import Button
+from kivy.graphics import Color, Ellipse, Line
+from kivy.uix.floatlayout import FloatLayout
+from functools import partial
+global X, Y
+#Initial coords.
+X = 400
+Y = 400
+
+class Canvas(Widget):
```

Comment:
`X` and `Y` variables seems to only be used in this class. Instead of having them as global variables, you could declare them as part of the Canvas object.

```python
def __init__(self, **kwargs):
    self.Y = 400
    self.X = 400
```

Reply:
Will do. Left it as global because it was being used across classes initially and forgot to change it to one class only.